src/base/Cellar.sol

```diff
@@ -598,6 +690,59 @@ contract Cellar is ERC4626, Ownable, Multicall {
         _depositTo(holdingPosition, assets);
     }
 
+    function beforeWithdraw(
+        uint256 assets,
+        uint256,
+        address,
+        address
+    ) internal override {
+        // Need to check if assets is greater than the high watermark
+        // because if the performanceFee is set to zero, and all cellar shares are redeemed,
+        // if the cellar has earned any yield, assets will be greater than the high watermark.
+        // Becuase the high watermark is only updated when performance fees are minted.
+        uint256 highWatermark = feeData.highWatermark;
+        feeData.highWatermark = assets > highWatermark ? 0 : highWatermark - assets;
+    }
+
+    function deposit(uint256 assets, address receiver) public override returns (uint256 shares) {
+        uint256 _totalAssets = totalAssets();
+
+        _takePerformanceFees(_totalAssets);
+
+        // Check for rounding error since we round down in previewDeposit.
+        if ((shares = _convertToShares(assets, _totalAssets)) == 0) revert USR_ZeroShares();
+
+        beforeDeposit(assets, shares, receiver);
```

Comment:
Seems like you could keep things DRY and replace lines 715-722 with `mint(shares, receiver)`?

Reply:
Same opportunity for `redeem`/`withdraw` (maybe a little more work there, you might have to create a shared internal function to prevent repeated calls to `_getData`)